src/Features/Core/Portable/UseAutoProperty/AbstractUseAutoPropertyAnalyzer.cs

```diff
@@ -0,0 +1,293 @@
+            {
+                return;
+            }
+
+            var semanticModel = symbolContext.Compilation.GetSemanticModel(propertyDeclaration.SyntaxTree);
+            var getterField = GetGetterField(semanticModel, property.GetMethod, cancellationToken);
+            if (getterField == null)
+            {
+                return;
+            }
+
+            // If the user made the field readonly, we only want to convert it to a property if we
+            // can keep it readonly.
+            if (getterField.IsReadOnly && !SupportsReadOnlyProperties(symbolContext.Compilation))
+            {
+                return;
+            }
+
+            if (!containingType.Equals(getterField.ContainingType))
+            {
+                // Field and property have to be in the same type.
+                return;
+            }
+
+            // Property and field have to agree on type.
+            if (!property.Type.Equals(getterField.Type))
+            {
+                return;
+            }
+
+            // Don't want to remove constants.
+            if (getterField.IsConst)
+            {
+                return;
+            }
+
+            if (getterField.DeclaringSyntaxReferences.Length != 1)
+            {
+                return;
+            }
+
+            // Field and property should match in static-ness
+            if (getterField.IsStatic != property.IsStatic)
+            {
+                return;
+            }
+
+            // A setter is optional though.
+            var setMethod = property.SetMethod;
+            if (setMethod != null)
+            {
+                var setterField =  GetSetterField(semanticModel, containingType, setMethod, cancellationToken);
+                if (setterField != getterField)
+                {
+                    // If there is a getter and a setter, they both need to agree on which field they are 
+                    // writing to.
+                    return;
+                }
+            }
+
+            var fieldReference = getterField.DeclaringSyntaxReferences[0];
+            var variableDeclarator = fieldReference.GetSyntax(symbolContext.CancellationToken) as TVariableDeclarator;
+            if (variableDeclarator == null)
+            {
+                return;
+            }
+
+            var fieldDeclaration = variableDeclarator?.Parent?.Parent as TFieldDeclaration;
+            if (fieldDeclaration == null)
+            {
+                return;
+            }
+
+            // Can't remove the field if it has attributes on it.
+            if (getterField.GetAttributes().Length > 0)
+            {
+                return;
+            }
+
+            // Looks like a viable property/field to convert into an auto property.
+            analysisResults.Add(new AnalysisResult(property, getterField, propertyDeclaration, fieldDeclaration, variableDeclarator,
+                property.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat)));
```

Comment:
Do we want this to be fully qualified? In what scenario is the user going to need a fully qualified property to disambiguate?